packages/ckeditor5-engine/src/view/renderer.js

```diff
@@ -930,6 +930,7 @@ function addInlineFiller( domDocument, domParentOrArray, offset ) {
 function areSimilar( node1, node2 ) {
 	return isNode( node1 ) && isNode( node2 ) &&
 		!isText( node1 ) && !isText( node2 ) &&
+		node1.nodeType !== 8 && node2.nodeType !== 8 &&
```

Comment:
`Node.COMMENT_NODE` via https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType.

Reply:
Also: `DomCoverter` [already implements `isComment()`](https://github.com/ckeditor/ckeditor5/blob/ad3562a2d3b6d5a1e8de276e7f032371ba260d63/packages/ckeditor5-engine/src/view/domconverter.js#L802-L804). I think it should be extracted to `ckeditor5-utils/src/dom` (like `isNode()`, `isText()` and many others), tested, and then used both in `DomConverter` and `Renderer`.

Reply:
Do we really need to add a dependency and bunch of tests for quite straightforward, native, and atomic comparison: `node.nodeType !== Node.COMMENT_NODE`? To me, it seems like overhead.

Reply:
I figured since there are 2 places that would benefit from the new thing, then why not? We have `is*()` helpers for most of the native DOM object types anyway. There are even [tests for it already](https://github.com/ckeditor/ckeditor5/blob/ad3562a2d3b6d5a1e8de276e7f032371ba260d63/packages/ckeditor5-engine/tests/view/domconverter/domconverter.js#L172-L183). 

We could use `Object.prototype.toString.apply( obj ) == '[object Comment]';` to make sure it is document-agnostic (like in `isRange()`).

Reply:
Or... to make it faster, we could call `isNode()` first and then check the `#type`.

Reply:
...or not :D It looks like `isNode()` is buggy. A comment does not have a `#defaultView`, so `isNode()` will always return `false` even though `Comment` inherits from `Node` 

![](https://user-images.githubusercontent.com/1099479/84140599-2a2f3b00-aa52-11ea-8e03-5ddef5e8f5c9.png)

Reply:
Maybe let's have a follow-up for this whole story then 😛